src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -363,6 +367,30 @@ private void validateSchema(DatasetSpecificationModel schema, Errors errors) {
         }
     }
 
+    private void validateRegion(DatasetRequestModel datasetRequest, Errors errors) {
+        if (datasetRequest.getRegion() != null) {
+            if (datasetRequest.getCloudPlatform() == null) {
+                errors.rejectValue("region", "InvalidRegionForPlatform",
+                    "Cannot set a region when a cloudPlatform is not provided.");
+            }
+            boolean supported = false;
+            switch (datasetRequest.getCloudPlatform()) {
+                case GCP:
+                    supported = SUPPORTED_GOOGLE_REGIONS.contains(datasetRequest.getRegion().toLowerCase());
```

Comment:
`toLowerCase` is used here but not used above when `SUPPORTED_GOOGLE_REGIONS` is created. How do you know the region names are already lower case?

Reply:
They're always lower case because that's how we've defined the enum.

Reply:
But there is nothing in the code that enforces this, and the assumption is in code that's not next to the enum member definitions. If a non-lowercase region was added, this code would break and we wouldn't be aware of it. Moving the `toLower()` into `GoogleRegion` would isolate this assumption in the enum, which would prevent the code breaking.

Looking at the way this enum is used, it seems like what you want is `GoogleRegion.fromValue()` to call `toLower()` on its argument. Once you'e done that, this code can be written as
```
supported = GoogleRegion.fromValue(datasetRequest.getRegion()) != null;
```
That would also let you remove the same code assumption from `GoogleRegionConverter.fromValue()`.